adlfs/transfer.py

```diff
@@ -0,0 +1,368 @@
+# -*- coding: utf-8 -*-
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+High performance multi-threaded module to up/download
+
+Calls method in `core` with thread pool executor to ensure the network
+is used to its maximum throughput.
+
+Only implements upload and download of (massive) files and directory trees.
+"""
+from collections import namedtuple
+from concurrent.futures import ThreadPoolExecutor
+import logging
+import multiprocessing
+import os
+import pickle
+import threading
+import time
+import uuid
+
+from adlfs.utils import clamp
+
+MAX_THREADS = 64
```

Comment:
This was my suggestion. My thinking was, a user can always change a module-level constant like this at runtime (without changing the code), so it's a _default_ _default_.